src/socket.rs

```diff
@@ -19,13 +19,21 @@ use realm_io;
 use socket2::SockRef;
 use tokio::io;
 use tokio::net::TcpListener;
+use tokio::net::TcpSocket;
 use tracing::warn;
 
 #[cfg(target_os = "linux")]
 pub fn set_transparent(l: &TcpListener) -> io::Result<()> {
     SockRef::from(l).set_ip_transparent(true)
 }
 
+#[cfg(target_os = "linux")]
+pub fn set_freebind(sock: &TcpSocket) -> io::Result<()> {
+    let sock = SockRef::from(sock);
+    sock.set_ip_transparent(true)?;
```

Comment:
With this change, ztunnel must be running with `CAP_NET_ADMIN` privileges, whereas previously that wasn't necessary.

I think we need to make a decision on what behavior do we want, since right now it will fail at runtime.

If we want to require `CAP_NET_ADMIN` capabilities, then we should check if `IP_TRANSPARENT` works at startup, and bail out if it's not. We should also modify existing code paths that tries to enable transparent mode to fail then.

If we want to proxy traffic even when running in non-transparent mode, then we need to ignore errors when setting `IP_TRANSPARENT` and `IP_FREEBIND`.

Reply:
IMO we should allow either mode. We already do this for setting IP_TRANSPARENT on the listener

Reply:
Should we make this behavior configurable instead of opportunistic?

Reply:
> IMO we should allow either mode. We already do this for setting IP_TRANSPARENT on the listener

Should we have a configurable flag to indicate we use transparent mode or not?